eox_tenant/api/v1/viewsets.py

```diff
@@ -1,29 +1,451 @@
 """
 API v1 viewsets.
 """
+import logging
+
 from rest_framework import viewsets
 from rest_framework.parsers import JSONParser
 
 from eox_tenant.api.v1.permissions import EoxTenantAPIPermission
 from eox_tenant.api.v1.serializers import MicrositeSerializer, RouteSerializer, TenantConfigSerializer
 from eox_tenant.models import Microsite, Route, TenantConfig
 
+logger = logging.getLogger(__name__)
+
+
+class AlternativeFieldLookupMixin:
+    """
+    This mixin allows to get the corresponding object via the autogenerated id field (pk)
+    or the field specified in the 'alternative_lookup_field' attribute.
+
+    If the value sent in the URL is an identifier composed of digits only, then the
+    lookup_field will remain the same. Otherwise, the alternative_lookup_field will become
+    the lookup field for the object.
+
+    This class assumes that the 'lookup_field' attribute of the view is a field whose value
+    is composed of digits only and expects the 'alternative_lookup_field' value to contain
+    non-numeric characters.
```

Comment:
Since this deprecates the use of the regular ID I would suggest that we leave a comment here stating that this Mixin can be removed after some date.
I would put July 2023 as tentative. After that, then we can remove this and only use the alternative.

Reply:
@felipemontoya done
https://github.com/eduNEXT/eox-tenant/blob/04a47c59db77041252abca6c951b9c5cd236fa61/eox_tenant/api/v1/viewsets.py#L29-L32